public/js/components/Scopes.js

```diff
@@ -116,18 +124,26 @@ const Scopes = React.createClass({
   propTypes: {
     pauseInfo: ImPropTypes.map,
     loadedObjects: ImPropTypes.map,
-    loadObjectProperties: PropTypes.func
+    loadObjectProperties: PropTypes.func,
+    selectedFrame: PropTypes.map
   },
 
   displayName: "Scopes",
 
   getInitialState() {
-    return { scopes: getScopes(this.props.pauseInfo) };
+    const { pauseInfo, selectedFrame } = this.props;
+    return { scopes: getScopes(pauseInfo, selectedFrame) };
```

Comment:
what was the reason we used state here? could we have changed Scopes to use `componentShouldUpdate`